File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0004-lending/type/TBureauDataImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tBureauData"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TBureauDataImpl implements TBureauData {
        private java.math.BigDecimal creditScore;
        private Boolean bankrupt;

    public TBureauDataImpl() {
    }

    public TBureauDataImpl(Boolean bankrupt, java.math.BigDecimal creditScore) {
        this.setBankrupt(bankrupt);
        this.setCreditScore(creditScore);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Bankrupt")
    public Boolean getBankrupt() {
        return this.bankrupt;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Bankrupt")
    public void setBankrupt(Boolean bankrupt) {
        this.bankrupt = bankrupt;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("CreditScore")
    public java.math.BigDecimal getCreditScore() {
        return this.creditScore;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("CreditScore")
    public void setCreditScore(java.math.BigDecimal creditScore) {
        this.creditScore = creditScore;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0004-lending/translator/expected/python/dmn/type_/TBureauDataImpl.py
import typing
import decimal
import datetime
import time
import isodate

import type_.TBureauData


# Generated(value = {"itemDefinition.ftl", "tBureauData"})
class TBureauDataImpl(type_.TBureauData.TBureauData):
    def __init__(self, bankrupt: typing.Optional[bool] = None, creditScore: typing.Optional[decimal.Decimal] = None):
        type_.TBureauData.TBureauData.__init__(self)

        self.bankrupt = bankrupt
        self.creditScore = creditScore

    def __eq__(self, other: typing.Any) -> bool:
        return self.equalTo(other)

    def __hash__(self):
        return self.hashCode()

    def __str__(self) -> str:
        return self.asString()
